glvis.cpp

```diff
@@ -1551,6 +1555,8 @@ int main (int argc, char *argv[])
    // server mode, read the mesh and the solution from a socket
    if (input == INPUT_SERVER_MODE)
    {
+      GetMainThread();
```

Comment:
The call to `SDLMainLoop` below also calls `GetMainThread()` -- we can just move that call here, before the construction of `serverThread`.

Reply:
Actually, this does not work -- `SDLMainLoop` blocks and waits to dispatch events, so using `GetMainThread()` here is the right thing to do.

Reply:
Just add a comment like:
```suggestion
      // Make sure the singleton object returned by GetMainThread() is
      // initialized from the main thread.
      GetMainThread();
```

Reply:
> Actually, this does not work -- `SDLMainLoop` blocks and waits to dispatch events, so using `GetMainThread()` here is the right thing to do.

Yes, I know, I did not mean to do that.:wink: